pkg/tmpl/context_funcs.go

```diff
@@ -35,7 +42,7 @@ func (c *Context) createFuncMap() template.FuncMap {
 		"fetchSecretValue": fetchSecretValue,
 		"expandSecretRefs": fetchSecretValues,
 	}
-	if c.preRender {
+	if c.preRender || disableInsecureFeatures {
```

Comment:
I guess it might result in a better U/X if insecure functions like `exec` and `readFile` with `disableInsecureFeatures ` failed with explicit errors, instead of being silently ignored like this.

How about adding another `else if disableInsecureFeatures` block to replace `exec` and `readFile` implementations so that those functions return certain errors, like `Template function `exec` is disabled as insecure because you set DISABLE_INSECURE_FEATURES=false`?

Reply:
I was wondering the same thing, not sure which is better though. With explicit errors will not allow me to do dry-run easily, since I use `exec` and `readFile` mostly to get remote data. If `exec` is used in more advanced way, say in a control flow - ie. if on hostname A do this, else do that, explicit error is a better choice in this case IMO.

Tell me what you think is better @mumoshu.

Reply:
> With explicit errors will not allow me to do dry-run easily, since I use exec and readFile mostly to get remote data.

What do you mean by "dry-run" in this context? 🤔

I think I have never heard of a use-case around that, but perhaps it's that you run `helmfile template` with `DISABLE_INSECURE_FEATURES=true` and verify that it works without actually running `exec` or `readFile`?

Reply:
I think we'd better add another flag that is more explicit!
I think many variants of "dry-run" can be considered. One that calls exec and ReadFile but no modification on system and the cluster is made, one that skips insecure features, etc. Taking the common term "dry-run" for the specific use-case might not be ideal.

I do believe your use-case is valid though!
How about adding another flag or an envvar that looks like `--skip-insecure-template-functions` or `SKIP_INSECURE_TEMPLATE_FUNCTIONS` for yours?